ambry-clustermap/src/main/java/com.github.ambry.clustermap/CompositeClusterManager.java

```diff
@@ -31,6 +34,7 @@
  * and reports inconsistencies in the views from the two underlying cluster managers.
  */
 class CompositeClusterManager implements ClusterMap {
+  private static final Logger LOGGER = LoggerFactory.getLogger(CompositeClusterManager.class);
   final StaticClusterManager staticClusterManager;
   final HelixClusterManager helixClusterManager;
   final HelixClusterManagerMetrics helixClusterManagerMetrics;
```

Comment:
Let's add strict check for both `helixClusterManager` and `helixClusterManagerMetrics`.  That is, check they are `null` in all places where dereference is invoked.

Reply:
After fixing the constructor, all places where these are used are surrounded by a null check.